pypeit/slittrace.py

```diff
@@ -0,0 +1,613 @@
+        """
+        Write the :class:`SlitTraceSet` data to an
+        `astropy.io.fits.BinTableHDU`.
+
+        This is a basic wrapper for
+        :func:`pypeit.datamodel.DataContainer.to_hdu` that includes
+        the nominal :class:`pypeit.masterframe.MasterFrame` header
+        data.
+
+        Args:
+            hdr (`astropy.io.fits.Header`, optional):
+                Baseline header to add to all returned HDUs. The
+                relevant :class:`pypeit.masterframe.MasterFrame` and
+                :class:`SlitTraceSet` keywords are always
+                added/edited in this baseline header.
+            add_primary (:obj:`bool`, optional):
+                If False, the returned object is a simple :obj:`list`
+                object with a single element, the
+                `astropy.io.fits.BinTableHDU`_ with the
+                :class:`SlitTraceSet` data. If True, the method
+                constructs an `astropy.io.fits.HDUList` with a
+                primary HDU. The nominal
+                :func:`pypeit.masterframe.MasterFrame` header data is
+                included in all returned HDUs.
+
+        Returns:
+            :obj:`list`, `astropy.io.fits.HDUList`_: A list of HDUs,
+            where the type depends on the value of ``add_primary``.
+        """
+        # TODO: Use super here?
+        return datamodel.DataContainer.to_hdu(self, hdr=self.build_master_header(hdr=hdr),
+                                              add_primary=add_primary)
+
+    def _validate(self):
+        """
+        Validate the slit traces.
+        """
+        # Allow the object to be empty
+        if self.left is None or self.right is None:
+            return
+        if self.left.shape != self.right.shape:
+            raise ValueError('Input left and right traces should have the same shape.')
+        if self.left.ndim == 1:
+            # Object contains a single slit.  Expand the dimensions.
+            self.left = np.expand_dims(self.left, 1)
+            self.right = np.expand_dims(self.right, 1)
+
+        # Do the same for the tweaked traces. NOTE: Tweaked traces will
+        # only exist if they were read from an output file; i.e., the
+        # current init does not allow for the tweaked traces to be
+        # defined directly.
+        if self.left_tweak is not None:
+            if self.left_tweak.shape != self.right_tweak.shape:
+                # TODO: Shouldn't get here, but just in case the init changes...
+                raise ValueError('Tweaked left and right traces should have the same shape.')
+            if self.left_tweak.ndim == 1:
+                # Object contains a single slit.  Expand the dimensions.
+                self.left_tweak = np.expand_dims(self.left_tweak, 1)
+                self.right_tweak = np.expand_dims(self.right_tweak, 1)
+
+        self.nspec, self.nslits = self.left.shape
+
+        # Center is always defined by the original traces, not the
```

Comment:
This does seem odd.  I would think tweaked is better for both.
Thoughts @jhennawi ?

Reply:
I don't think center is used much, but I think it should probably be the tweaked edges. Maybe we can look for instances of center. I don't understand the point in the comment about the slit ids, since tweaking never changes the ordering or number of slits, and the objects are assigned to slits later.

Reply:
I'm not sure where they're used, but the slit IDs (see `_set_slitids`) are based on the pixel coordinate of the slit center.  That's the source of the comment.